src/pages/index.js

```diff
@@ -4,53 +4,136 @@ import PopupWithForm from "../components/PopupWithForm.js";
+  api
+    .editProfile({ name: data.name, description: data.description })
+    .then((res) => {
+      userInfo.setUserInfo({ name: res.name, description: res.about });
+      profilePopupForm.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+  profilePopupForm.setLoading(false);
   return;
 }
 
+function handleAvatarFormSubmit(data) {
+  avatarPopupForm.setLoading(true);
+  api
+    .updateAvatar(data.image)
+    .then((res) => {
+      avatarPopupForm.close();
+      avatarPopupForm.reset();
+      userInfo.setAvatar(data.image);
+    })
+    .catch((err) => {
+      console.error("Error saving avatar:", err);
+    })
+    .finally(avatarPopupForm.setLoading(false));
+}
+
 function handleAddProfileFormSubmit(data) {
-  const name = data.name;
-  const link = data.image;
-  const cardElement = createCard({ name, link });
-  layerSection.addItem(cardElement);
-  addCardPopupForm.close();
-  addCardPopupForm.reset();
-  addFormValidator.toggleButtonState();
+  addCardPopupForm.setLoading(true);
+  api
+    .addCard({
+      name: data.name,
+      link: data.image,
+    })
+    .then((res) => {
+      layerSection.addItem(createCard(res));
+      addCardPopupForm.close();
+      addCardPopupForm.reset();
+      addFormValidator.toggleButtonState();
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+  addCardPopupForm.setLoading(false);
   return;
 }
 
 function createCard(data) {
-  const card = new Card(data, "#card-template", handleCardImageClick);
+  const card = new Card(
+    data,
+    "#card-template",
+    handleCardImageClick,
+    handleCardDelete,
+    handleLikeCard,
+    handleDeleteModal
+  );
   return card.renderCard();
 }
 
 function handleCardImageClick(name, link) {
   cardImagePopup.open({ name, link });
 }
 
+function handleCardDelete(card) {
+  api
+    .deleteCard(card._id)
+    .then(() => {
+      card._cardElement.remove();
+      card._cardElement = null;
+      deletePopup.close();
+    })
+    .catch((err) => {
+      console.error("Failed to delete card:", err);
+    });
+}
+
+function handleDeleteModal(card) {
+  deletePopup.open(card);
+}
+
+function handleLikeCard(card) {
+  if (card._isLiked === false) {
+    return api
+      .addLike(card._id)
+      .then(() => {
+        card._isLiked = true;
+        card.addLikeButton();
+      })
+      .catch((err) => {
+        console.error("Error on liking card", err);
+      });
+  } else if (card._isLiked === true) {
+    return api
+      .removeLike(card._id)
+      .then(() => {
+        card._isLiked = false;
+        card.removeLikeButton();
+      })
+      .catch((err) => {
+        console.error("Error on unliking", err);
+      });
+  }
+}
+
 //Image srcs
-const avatar = document.getElementById("profile-image");
-avatar.src = jacquesCousteau;
 const logo = document.getElementById("logo");
 logo.src = logoImage;
+
 //Class Instances
 const addFormValidator = new FormValidator(settings, "#modal-add-form");
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object: `formValidators`.  Then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // Here you get the name of the form (if you don’t have it then you need to add it into each form in `index.html` first)
    const formName = formElement.getAttribute('name')

   // Here you store the validator using the `name` of the form
    formValidators[formName] = validator;
    validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```

Reply:
This one was tricky and required quite a bit of refactoring. In regards to naming the forms, I used their popup names. Would it be better to call them validators?
Refactoring complete and is working.

Reply:
> Would it be better to call them validators

No, everything is fine now


The forms should be named as forms